### DISCUSSION TITLE
How to exclude directories?

### DISCUSSION THREAD

**Discussion by Cicorione** [speedata/publisher#533] [Q&A]
Hi @pgundlach 

Is there way to pass to Publisher a list of directories to ignore during the compilation process?

Perhaps it would be easier, for instance, if using a tilde (`~`) Publisher would ignore automatically any files or directories that start with it:
```
~proof
~test
chapters
images
~collection.xml
~layout-proof.xml
~pic1.jpg
pic2.jpg
data.xml
layout.xml
```

Thank you! 🙏

**Comment by pgundlach** [comment]
You should use `--no-local` as an argument to sp and add the directories that should be included with `--extra-dir`.  You can use the extra-dir in the configuration, and I think about adding the no-local switch to the configuration as well.

I cannot do the tilde thing because this might break existing layouts (although unlikely). 

Otherwise I don't have a solution. It should be possible to create an option to pass a list of to-ignore-files, but I doubt that this is very useful

**Reply by Cicorione** [reply]
I see...

On my end if I use no `--no-local` Publisher doesn't go inside each section sub-directories; by the other hand I have plenty of directories everywhere, and although do not affect the PDF creation however are delaying the process. 

> I cannot do the tilde thing because this might break existing layouts (although unlikely).

Even if you set this as **experimental** feature and not as default?
For me is just `~` = `skip` but I can understand your concern.

Thanks.

**Comment by pgundlach** [comment]
> I see...
> 
> On my end if I use no `--no-local` Publisher doesn't go inside each section sub-directories; by the other hand I have plenty of directories everywhere, and although do not affect the PDF creation however are delaying the process.

have you tried using `--extra-dir` for these directories? They are added to the recursive file search. 

 
> > I cannot do the tilde thing because this might break existing layouts (although unlikely).
> 
> Even if you set this as **experimental** feature and not as default? For me is just `~` = `skip` but I can understand your concern.

well, every experimental feature will become a standard feature (or removed) so this should have yet another switch, which I am always recultant to add.

**Reply by Cicorione** [reply]
Actually I did not, I have 21 sections (directories) and I dislike this organization. I hope at the end to move everything inside one "products" directory, I already have several duplicates and I hate this. 

Unfortunately my supervisors still don't get how all these `Speedata Publisher` works so I prefer keep the organization the way they understand; but I am pretty sure when the catalog is going to have a better shape, they will understand that is better have everything in one place and use the xml as reference for the sections' catalog.

**Reply by pgundlach** [reply]
I assume you know that you can add `--extra-dir` multiple times? Also provide the setting in the configuration file.

**Reply by Cicorione** [reply]
No, I don't. There is too much mess, I had better to clean up a bit instead. Anyway thanks for all your advice.

**Comment by Cicorione** [comment]
Hi @pgundlach just for the sake of discussion...

But what about if you in the global configuration can decide to assign a special (and single) **"skip" character** to any file/directory which by default is not assigned, this won't break anything I guess.

**Reply by pgundlach** [reply]
Let me think about it. I am a bit reluctant as it introduces yet another layer of configuration which is a bit obscure and hard to debug...

**Reply by Cicorione** [reply]
This can lie on your thoughts as much as yo wish!
